topydo/lib/ChangeSet.py

```diff
@@ -99,7 +99,10 @@ def save(self, p_todolist):
 
         current_hash = hash_todolist(p_todolist)
         list_todo = (self.todolist.print_todos()+'\n').splitlines(True)
-        list_archive = (self.archive.print_todos()+'\n').splitlines(True)
+        try:
+            list_archive = (self.archive.print_todos()+'\n').splitlines(True)
+        except AttributeError:
+            list_archive = []
```

Comment:
By looking at the code, this would erase the archive file:

```sh
topydo -a do 1
topydo revert
```

If so, my suggestion is to use a copy of the archive content from the previous backup, or the content of the current done.txt if copy is also missing in the changeset.